client/my-sites/woocommerce/woop/landing-page.tsx

```diff
@@ -9,29 +10,60 @@ import Image04 from 'calypso/assets/images/woocommerce/woop-cta-image04.jpeg';
 import CtaSection from 'calypso/components/cta-section';
 import FixedNavigationHeader from 'calypso/components/fixed-navigation-header';
 import MasonryWave from 'calypso/components/masonry-wave';
+import { addQueryArgs } from 'calypso/lib/url';
 
 interface Props {
 	startSetup: () => void;
+	isFeatureActive: boolean;
+	upgradingPlan: Record< string, unknown >;
+	siteSlug: string;
 }
 
 const images = [ { src: Image01 }, { src: Image02 }, { src: Image03 }, { src: Image04 } ];
 
-const WoopLandingPage: React.FunctionComponent< Props > = ( { startSetup } ) => {
+const WoopLandingPage: React.FunctionComponent< Props > = ( {
+	startSetup,
+	siteSlug,
+	isFeatureActive,
+	upgradingPlan,
+} ) => {
+	const { __ } = useI18n();
 	const navigationItems = [ { label: 'WooCommerce', href: `/woocommerce-installation` } ];
 	const ctaRef = useRef( null );
 
+	function onCTAHandler() {
+		if ( ! isFeatureActive && upgradingPlan?.product_slug ) {
+			return ( window.location.href = addQueryArgs(
+				{ redirect_to: window.location.href },
+				`/checkout/${ siteSlug }/${ upgradingPlan.product_slug }`
+			) );
+		}
+
+		startSetup();
+	}
+
+	const upgradeButtonText = sprintf(
+		/* translators: %s: The upgrading plan name (ex.: WordPress.com Business) */
+		__( 'Upgrade to %s' ),
+		upgradingPlan.product_name
+	);
+
 	return (
 		<div className="woop__landing-page">
 			<FixedNavigationHeader navigationItems={ navigationItems } contentRef={ ctaRef }>
-				<Button onClick={ startSetup } primary>
-					{ translate( 'Set up my store!' ) }
+				<Button onClick={ onCTAHandler } primary>
+					{ isFeatureActive ? __( 'Set up my store!' ) : upgradeButtonText }
 				</Button>
 			</FixedNavigationHeader>
 			<CtaSection
-				title={ translate( 'Have something to sell?' ) }
-				headline={ translate( 'Build exactly the eCommerce website you want.' ) }
-				buttonText={ translate( 'Set up my store!' ) }
-				buttonAction={ startSetup }
+				title={ __( 'Have something to sell?' ) }
+				headline={
+					isFeatureActive
+						? __( 'Build exactly the eCommerce website you want.' )
+						: __( 'Upgrade to the Premium plan and set up your WooCommerce store.' )
```

Comment:
Looks like a Premium reference snuck in here:
<img width="1178" alt="Screen Shot 2021-11-17 at 2 33 10 PM" src="https://user-images.githubusercontent.com/1689238/142270644-679a957b-6551-4404-acb2-41b501c33a88.png">

Reply:
Awesome catch!

Addressed [here](https://github.com/Automattic/wp-calypso/pull/58173/commits/c8663610c2e7863a3df16b279d69d43464059006)

![image](https://user-images.githubusercontent.com/77539/142271152-c3704cfa-2425-4243-b4fe-bc194b165210.png)

Reply:
it wasn't an awesome catch, I'd say it's obvious actually. I didn't see it. 😅 Thanks.